File: DarkOrbit 7.5.3/Net/netty/handlers/PetRequestHandler.cs
using Ow.Game;
using Ow.Managers;
using Ow.Net.netty.commands;
using Ow.Net.netty.requests;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ow.Net.netty.handlers
{
    class PetRequestHandler : IHandler
    {
        public void execute(GameSession gameSession, byte[] bytes)
        {
            var read = new PetRequest();
            read.readCommand(bytes);

            var player = gameSession.Player;
            if (player.RankId != 21) return;

            switch (read.petRequestType)
            {
                case PetRequest.LAUNCH:
                    player.Pet.Activate();
                    break;
                case PetRequest.TOGGLE_ACTIVATION:
                    player.Pet.Activate();
                    break;
                case PetRequest.DEACTIVATE:
                    player.Pet.Deactivate();
                    break;
                case PetRequest.HOTKEY_GUARD_MODE:
                    player.Pet.SwitchGear(PetGearTypeModule.GUARD);
                    break;
                case PetRequest.HOTKEY_REPAIR_SHIP:
                    //player.Pet.ComboShipRepair();
                    break;
            }
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/handlers/PetRequestHandler.java
package simulator.netty.handlers;

import net.game_server.GameServerClientConnection;
import net.game_server.GameSession;
import simulator.netty.ClientCommand;
import simulator.netty.clientCommands.PetRequest;
import simulator.users.Account;
import utils.Log;

/**
 Created by LEJYONER on 21/09/2017.
 */

public class PetRequestHandler
        implements ICommandHandler {

    private final GameServerClientConnection mGameServerClientConnection;
    private final PetRequest 			 	 mPetRequest;
    
    public PetRequestHandler(final GameServerClientConnection pGameServerClientConnection,
                                          final ClientCommand pPetRequest) {
        this.mGameServerClientConnection = pGameServerClientConnection;
        this.mPetRequest = (PetRequest) pPetRequest;
        
    }

    @Override
    public void execute() {
    	Log.pt("ACTIVE PET REQUEST HANDLER");
        final GameSession gameSession = this.mGameServerClientConnection.getGameSession();
        if (gameSession != null) {
        	final Account account = gameSession.getAccount();       	
        	if(account.isAdmin() || account.havePet()) {
        		switch(mPetRequest.petRequestType) {
        			case PetRequest.LAUNCH:
        				account.getPetManager().Activate();
        				break;
        			case PetRequest.TOGGLE_ACTIVATION:
        				account.getPetManager().Activate();
        				break;
        			case PetRequest.DEACTIVATE:
        				account.getPetManager().Deactivate();
        				break;
        			case PetRequest.HOTKEY_GUARD_MODE:
        				account.getPetManager().GuardMode();
        			case PetRequest.HOTKEY_REPAIR_SHIP:
        				account.getPetManager().ComboShipRepair();
        				break;
        		}
        	}
        }
    }
}
